cli/src/main/java/org/jboss/sbomer/cli/feature/sbom/command/CycloneDxGenerateOperationCommand.java

```diff
@@ -111,7 +112,15 @@ protected Path doGenerate() {
             throw new ApplicationException("Unable to create a new Bom");
         }
 
-        String deliverableUrl = config.getDeliverableUrls().get(getParent().getIndex());
+        String deliverableUrl;
+        try {
+            String url = config.getDeliverableUrls().get(getParent().getIndex());
+            deliverableUrl = URI.create(url).normalize().toURL().toString();
+        } catch (MalformedURLException e) {
+            String url = config.getDeliverableUrls().get(getParent().getIndex());
+            throw new ApplicationException("Could not parse deliverable URL " + url, e);
```

Comment:
Theoretically this should never happen, if this is run on successful operations.